### PR TITLE
Lpd 37967 pr 790

### PR DESCRIPTION
**Pull request by slnn**
More SF and sort for these new inserts according to the requirement in [LPD-37967 The insertion of the new tables should be sorted unless order really matters. Please fix the SQL in custom_object_definition and system_object_definition (may be other affected FTL as well. But sort whatever inserts can besorted)](https://github.com/slnn/liferay-portal/pull/790/commits/9daabb2444af36c2d2e14063c409c5bb847e9ecc )

Hi @@devpedrotavares

Please double check if my changes break anything, thanks!

Lily

### CODE REVIEW

**Comment by devpedrotavares** [comment]
> More SF and sort for these new inserts according to the requirement in [LPD-37967 The insertion of the new tables should be sorted unless order really matters. Please fix the SQL in custom_object_definition and system_object_definition (may be other affected FTL as well. But sort whatever inserts can besorted)](https://github.com/slnn/liferay-portal/pull/790/commits/9daabb2444af36c2d2e14063c409c5bb847e9ecc)
> 
> Hi @@devpedrotavares
> 
> Please double check if my changes break anything, thanks!
> 
> Lily

Hi @slnn , all is working as expected, thank you very much for working on this!

**Comment by devpedrotavares** [comment]
Hi @slnn , is there something else needed to forward this? Thank you

**Comment by slnn** [comment]
> Hi @slnn , is there something else needed to forward this? Thank you

Nothing from me, I have asked @dantewang to review this

**Comment by devpedrotavares** [comment]
> > Hi @slnn , is there something else needed to forward this? Thank you
> 
> Nothing from me, I have asked @dantewang to review this

Okay, thank you!

**Comment by dantewang** [comment]
@slnn What is the rule of the sorting? Are they sorted alphabetically? Based on table name? Object name? DataFactory method name?

**Comment by slnn** [comment]
> @slnn What is the rule of the sorting? Are they sorted alphabetically? Based on table name? Object name? DataFactory method name?

 Based on table name and  sorted alphabetically

https://github.com/slnn/liferay-portal/pull/790/commits/9daabb2444af36c2d2e14063c409c5bb847e9ecc

**Comment by dantewang** [comment]
@devpedrotavares @slnn I think there's a naming issue. We need to be able to tell the table names directly from the variable/method names to verify and follow the sorting, but some variables and methods have names that don't directly show the model/table name. For example, `relationshipObjectFieldModel` is inserted into `ObjectField` table but the name starts with `r`; `getInsertIntoDynamicObjectDefinitionTable()` creates sqls inserting into `ObjectDefinition` but the name looks like it's inserting into `DynamicObjectDefinition`; there's another method about `DynamicExtensionObjectDefinition` but the table name is actually `ObjectDefinition_x`.

Please try to rename to make the things more recognizable for sorting.

**Comment by slnn** [comment]
OK, got it

> @devpedrotavares @slnn I think there's a naming issue. We need to be able to tell the table names directly from the variable/method names to verify and follow the sorting, but some variables and methods have names that don't directly show the model/table name. For example, `relationshipObjectFieldModel` is inserted into `ObjectField` table but the name starts with `r`; `getInsertIntoDynamicObjectDefinitionTable()` creates sqls inserting into `ObjectDefinition` but the name looks like it's inserting into `DynamicObjectDefinition`; there's another method about `DynamicExtensionObjectDefinition` but the table name is actually `ObjectDefinition_x`.
> 
> Please try to rename to make the things more recognizable for sorting.

**Comment by slnn** [comment]
ci:test:sf

**Comment by slnn** [comment]
Hi @@devpedrotavares

Please double check if my changes break anything, thanks!

Lily

File: modules/util/portal-tools-sample-sql-builder/src/main/resources/com/liferay/portal/tools/sample/sql/builder/dependencies/system_object_definitions.ftl [hunk 1]
```diff
@@ -4,22 +4,22 @@
...
-	<#if systemObjectDefinitionModel.getDBTableName() == "CommerceOrder">
-		 ${dataFactory.toInsertSQL(dataFactory.newObjectActionModel(systemObjectDefinitionModel.objectDefinitionId, notificationTemplateModel.notificationTemplateId))}
+	<#if objectDefinitionModel.getDBTableName() == "CommerceOrder">
+		 ${dataFactory.toInsertSQL(dataFactory.newObjectActionModel(objectDefinitionModel.objectDefinitionId, notificationTemplateModel.notificationTemplateId))}
 	</#if>
 
-	<#if systemObjectDefinitionModel.getDBTableName() == "User_">
-		${dataFactory.getExtensionDynamicObjectDefinitionTableCreateSQL(systemObjectDefinitionModel)}
+	<#if objectDefinitionModel.getDBTableName() == "User_">
+		${dataFactory.getExtensionDynamicObjectDefinitionTableCreateSQL(objectDefinitionModel)}
```
Comment (dantewang) [inline modules/util/portal-tools-sample-sql-builder/src/main/resources/com/liferay/portal/tools/sample/sql/builder/dependencies/system_object_definitions.ftl:17]: @slnn Please also consider renaming these methods (create and insert). Make the name reflect the table name.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by devpedrotavares** [comment]
Hi @slnn @dantewang , 
some changes were recently merged to master and caused this not to work as expected. I sent a follow up [PR](https://github.com/liferay-core-infra/liferay-portal/pull/7664) with the adjustment needed.

File: modules/util/portal-tools-sample-sql-builder/src/main/resources/com/liferay/portal/tools/sample/sql/builder/dependencies/system_object_definitions.ftl (hunk 1 above)
Comment (devpedrotavares) [reply modules/util/portal-tools-sample-sql-builder/src/main/resources/com/liferay/portal/tools/sample/sql/builder/dependencies/system_object_definitions.ftl:17]: Hi @dantewang , the method `getExtensionDynamicObjectDefinitionTableCreateSQL` is used to dynamically create a table for the object definition. It's being used for more than one object definition. So the name of the table varies.

Comment (devpedrotavares) [reply modules/util/portal-tools-sample-sql-builder/src/main/resources/com/liferay/portal/tools/sample/sql/builder/dependencies/system_object_definitions.ftl:17]: hi @dantewang , 

I'm not sure that I understand your point. But if I can contribute, the method `getExtensionDynamicObjectDefinitionTableCreateSQL` will generate the create SQL for a dynamic object definition table, so table name will vary

**Pull request closed without merging**